napari/_qt/qt_chunk_receiver.py

```diff
@@ -0,0 +1,48 @@
+from qtpy.QtCore import QObject, Signal
+
+from ..layers.base import Layer
+from ..utils.chunk import ChunkRequest, chunk_loader
+from ..utils.perf import add_instant_event
+
+
```

Comment:
This `ChunkLoader` specific class seems like it could be turned into something totally generic. Some sort of thing that maps our events to Qt slots. Maps workers thread events to GUI events. Probably not worth working make a generic thing. But it is pointing towards some type of "call this event but in the GUI thread" type of generic thing.